x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -6,64 +6,118 @@
+import { getDefaultSingleMetricViewerPanelTitle } from './get_default_panel_title';
 
 export interface SingleMetricViewerInitializerProps {
   bounds: TimeRangeBounds;
-  defaultTitle: string;
   initialInput?: Partial<SingleMetricViewerEmbeddableInput>;
-  job: MlJob;
-  onCreate: (props: Partial<SingleMetricViewerEmbeddableUserInput>) => void;
+  mlApiServices: MlApiServices;
+  onCreate: (props: SingleMetricViewerEmbeddableUserInput) => void;
   onCancel: () => void;
 }
 
 export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
-  defaultTitle,
   bounds,
   initialInput,
-  job,
   onCreate,
   onCancel,
+  mlApiServices,
 }) => {
-  const isNewJob = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
+  const isMounted = useMountedState();
+  const newJobUrl = useMlLink({ page: ML_PAGES.ANOMALY_DETECTION_CREATE_JOB });
+  const [jobIds, setJobIds] = useState(initialInput?.jobIds ?? []);
 
-  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const initialValuesRef = useRef({
+    titleManuallyChanged: !!initialInput?.title,
+    isNewJob:
+      initialInput?.jobIds !== undefined && jobIds.length && initialInput?.jobIds[0] !== jobIds[0],
+  });
+
+  const [job, setJob] = useState<MlJob | undefined>();
+  const [panelTitle, setPanelTitle] = useState<string>(initialInput?.title ?? '');
   const [functionDescription, setFunctionDescription] = useState<string | undefined>(
     initialInput?.functionDescription
   );
   // Reset detector index and entities if the job has changed
   const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(
-    !isNewJob && initialInput?.selectedDetectorIndex ? initialInput.selectedDetectorIndex : 0
+    !initialValuesRef.current.isNewJob && initialInput?.selectedDetectorIndex
+      ? initialInput.selectedDetectorIndex
+      : 0
   );
   const [selectedEntities, setSelectedEntities] = useState<MlEntity | undefined>(
-    !isNewJob && initialInput?.selectedEntities ? initialInput.selectedEntities : undefined
+    !initialValuesRef.current.isNewJob && initialInput?.selectedEntities
+      ? initialInput.selectedEntities
+      : undefined
   );
-
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
   const isPanelTitleValid = panelTitle.length > 0;
 
+  useEffect(
+    function setUpPanel() {
+      if (isMounted()) {
+        async function fetchJob() {
+          const { jobs } = await mlApiServices.getJobs({ jobId: jobIds.join(',') });
+
+          if (jobs.length > 0) {
+            // Reset values if the job has changed
+            if (initialValuesRef.current.isNewJob) {
+              setSelectedDetectorIndex(0);
+              setSelectedEntities(undefined);
+              setFunctionDescription(undefined);
+            }
+
+            setJob(jobs[0]);
+            setErrorMessage(undefined);
+          }
+        }
```

Comment:
I don't think you should reset values right after fetch, because it could also be the initial fetch. 
Instead, it should happen instantly on the `jobIds` selection change.

Reply:
Good call - simplified handling of job change in 5dd476550c2fe02731034b2ca1d5d59c58863d2a